internal/core/model/adapters.rs

```diff
@@ -288,13 +338,10 @@ where
         let end = mapping.binary_search(&(index + count)).unwrap_or_else(|e| e);
         let range = start..end;
 
-        if !range.is_empty() {
```

Comment:
The bug was here.  We should also adapt the count after the end point if the removed rows were already filtered.  (and i also simplified the code)
A similar problem existed in row_added.

Reply:
Ahh that's where. Makes sense - and nice simplification!

Reply:
Looks like the C++ implementation didn't have this bug - and now they look also much more alike :)

Reply:
The C++ implementation still has the bug in the ReverseModel row_removed